app/helpers/i18n_helper.rb

```diff
@@ -13,4 +13,20 @@ def set_locale
 
     I18n.locale = spree_current_user.andand.locale || cookies[:locale] || I18n.default_locale
   end
+
+  def valid_locale(object_with_locale)
+    if object_with_locale.present? &&
+       object_with_locale[:locale].present? &&
+       available_locale?(object_with_locale[:locale])
+      object_with_locale[:locale]
```

Comment:
I think I'm missing something. Why do we use `[:locale]` instead of `.locale`? As far as I see the object passed is always a user object.

Reply:
[] works for objects and hashes, right? this way it also works for hashes and, for example, I can write the tests without a user object, only a hash.

Reply:
Isn't that YAGNI?

Reply:
I was just being lazy, instead of using a user obj to test I wanted to test with a hash with only.
Anyway, I changed it, pls have a look.